plugins/flutter_aepedge/test/flutter_aepedge_test.dart

```diff
@@ -44,4 +46,148 @@ void main() {
+
+    final List<EventHandle> expectedResponse = [expectedEvent];
+    final List<dynamic> returnedResponse = [returnedEvent.data];
+
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return returnedResponse;
+      });
+    });
+    test('invokes correct method', () async {
+      await Edge.sendEvent(experienceEvent);
+
+      expect(log, <Matcher>[
+        isMethodCall(
+          'sendEvent',
+          arguments: experienceeventData,
+        ),
+      ]);
+    });
+
+    test('returns correct result', () async {
+      final actualEventHandleResponse = await Edge.sendEvent(experienceEvent);
+      expect(actualEventHandleResponse[0].payload, expectedResponse[0].payload);
+      expect(actualEventHandleResponse[0].type, expectedResponse[0].type);
+    });
+  });
+
+  group('sendEvent nested', () {
+    final Map<dynamic, dynamic> mapValue = {"keySample": "keyValue"};
+    final Map<dynamic, dynamic> dataValue = {"keySample1": "keyValue1"};
+    final Map<dynamic, dynamic> xdmData = {"eventType": mapValue};
+    final Map<String, dynamic> data = {"free": "form", "data": dataValue};
+
+    final Map<String, dynamic> experienceeventData = {
+      "xdmData": xdmData,
+      "data": data,
+      "datasetIdentifier": "datasetExample",
+    };
+
+    //setup experienceEvent
+    final ExperienceEvent experienceEvent =
+        ExperienceEvent(experienceeventData);
+
+    final String eventHandleType = "state:store";
+    final List<dynamic> eventHandlePayload = [
+      {"maxAge": 1800, "value": "testValue1", "key": "keyExample1"},
+      {"maxAge": 34128000, "value": "testValue2", "key": "keyExample2"}
+    ];
+
+    final Map<dynamic, dynamic> eventHandleExpectedData = {
+      "type": eventHandleType,
+      "payload": eventHandlePayload
+    };
+
+    final String eventHandleType2 = "state:store";
+    final List<dynamic> eventHandlePayload2 = [
+      {
+        "type": "samplestore#",
+        "destinationId": "345",
+        "alias": "auto",
+        "segments": [
+          {"id": "00001234-7nfj-78u0-0ne7-nju348098jd1"},
+          {"id": "78947208-yuj6-78nh-672m-k8792d7v9wnb"}
+        ]
+      },
+    ];
+
+    final Map<dynamic, dynamic> eventHandleExpectedData2 = {
+      "type": eventHandleType2,
+      "payload": eventHandlePayload2
+    };
+
+    //setup returned and expected EventHandles
+    final EventHandle expectedEvent = EventHandle(eventHandleExpectedData);
+    final EventHandle returnedEvent = EventHandle(eventHandleExpectedData);
+
+    final EventHandle expectedEvent2 = EventHandle(eventHandleExpectedData2);
+    final EventHandle returnedEvent2 = EventHandle(eventHandleExpectedData2);
+
+    final List<EventHandle> expectedResponse = [expectedEvent, expectedEvent2];
+    final List<dynamic> returnedResponse = [
+      returnedEvent.data,
+      returnedEvent2.data
+    ];
```

Comment:
same comment here, can we clean these up and only keep expected* and actual* variables?

Reply:
I need to return "List\<dynamic\>" returnedResponse for setMockMethod.  It doesn't take "List\<EventHandle\>".

Reply:
hmm.. I would expect to mock with an object with same type as the promise type. What error do you see?

-----
Unrelated, but looking into this found that [MethodChannel#setMockMethodCallHandler](https://docs.flutter.dev/release/breaking-changes/mock-platform-channels) is actually deprecated, and the new recommended way of mocking is WidgetTester 
```
tester.binding.defaultBinaryMessenger.setMockDecodedMessageHandler(myChannel, ...);
```
Defining a handler allows you to mock all the platform methods and declare the return type. Might be helpful especially if a test involves multiple APIs
```
import 'package:flutter/services.dart'; 
import 'package:flutter_test/flutter_test.dart';

void mockPackageInfo() {
  const channel = MethodChannel('plugins.flutter.io/package_info');

  handler(MethodCall methodCall) async {
    if (methodCall.method == 'getAll') {
      return <String, dynamic>{
        'key': 'value'
      };
    }
    return null;
  }

  TestWidgetsFlutterBinding.ensureInitialized();

  TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
      .setMockMethodCallHandler(channel, handler);
}
```
we should discuss about this in the main repo.

Reply:
The error I saw:
```
 PlatformException(error, Invalid argument: Instance of 'EventHandle', null, null)
  package:flutter/src/services/message_codecs.dart 607:7     StandardMethodCodec.decodeEnvelope
  package:flutter/src/services/platform_channel.dart 177:18  MethodChannel._invokeMethod
```

Reply:
I will write an research issue about setMockMethodCallHandler, so we can follow that up.

Reply:
On the StandardMethodCodec error -> thanks, that makes sense since the returned value from the platform is of standard message type (Map) and it was not cast yet to List<EventHandle> (happening in dart -> sendEvent).
This was helpful - https://api.flutter.dev/flutter/services/StandardMessageCodec-class.html | https://api.flutter.dev/flutter/services/StandardMethodCodec-class.html

That being said I think we can just simplify the implementation a bit, eg:

remove expectedEvent, expectedEvent2 and returnedResponse and construct in-line:
```
final List<EventHandle> expectedResponse = [EventHandle(expectedEventHandle), EventHandle(expectedEventHandle2)];
...
setUp(() {
      channel.setMockMethodCallHandler((MethodCall methodCall) async {
        log.add(methodCall);
        return [expectedEventHandle, expectedEventHandle2];
      });
    });